orte/mca/ess/base/ess_base_std_prolog.c

```diff
@@ -56,6 +59,10 @@ int orte_ess_base_std_prolog(void)
             goto error;
         }
     }
+    /* check for OPAL_PREFIX */
+    if (NULL != (evar = getenv("OPAL_PREFIX"))) {
+        opal_setenv("PMIX_PREFIX", evar, false, &environ);
```

Comment:
we can fix the first case by moving this down into opal/mca/pmix/pmix2x itself (note that you can't put it in the ORTE code as that wouldn't help the direct launch situation). There is no way to fix the other case - the user will just have to provide PMIX_PREFIX in that scenario.